pkg/kubectl/cmd/edit.go

```diff
@@ -0,0 +1,419 @@
+package cmd
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util/editor"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util/jsonmerge"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	//	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	//	"github.com/openshift/origin/pkg/cmd/util/editor"
+	//"github.com/openshift/origin/pkg/util/jsonmerge"
+)
+
+const (
+	editLong = `Edit a resource from the default editor.
+
+The edit command allows you to directly edit any API resource you can retrieve via the
+command line tools. It will open the editor defined by your OSC_EDITOR, GIT_EDITOR,
```

Comment:
We should probably nuke KUBE_EDITOR and tell people to set EDITOR.  I only
copied that pattern from GIT, but in practice anyone who wants it can just
set it on their command line.

On Tue, Jun 30, 2015 at 12:51 PM, Brian Grant <notifications@github.com>
wrote:

> In pkg/kubectl/cmd/edit.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/10502#discussion_r33595301>
> :
>
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util/editor"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util/jsonmerge"
> > +
> > +	"github.com/golang/glog"
> > +	"github.com/spf13/cobra"
> > +
> > +	//	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
> > +	//	"github.com/openshift/origin/pkg/cmd/util/editor"
> > +	//"github.com/openshift/origin/pkg/util/jsonmerge"
> > +)
> > +
> > +const (
> > +	editLong = `Edit a resource from the default editor.
> > +
> > +The edit command allows you to directly edit any API resource you can retrieve via the
> > +command line tools. It will open the editor defined by your OSC_EDITOR, GIT_EDITOR,
>
> We shouldn't use OSC_EDITOR -- that's openshift. We should use KUBE_EDITOR.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/10502/files#r33595301>
> .
>



-- 
Clayton Coleman | Lead Engineer, OpenShift